src/app/applications/datatarget/datatarget-test-connection/datatarget-test-connection.component.ts

```diff
@@ -0,0 +1,129 @@
+  private applicationId: number;
+  private dataTargetId: number;
+  private iotDeviceId: number;
+  private subscriptions = [];
+
+  constructor(
+    private deviceService: IoTDeviceService,
+    private dataTargetService: DatatargetService,
+    private route: ActivatedRoute,
+    private payloadDecoderService: PayloadDecoderService,
+    private payloadDeviceDataTargetService: PayloadDeviceDatatargetService,
+    public applicationService: ApplicationService
+  ) {}
+
+  ngOnInit(): void {
+    this.applicationId = +this.route.parent.snapshot.paramMap.get("id");
+    this.dataTargetId = +this.route.parent.snapshot.paramMap.get("datatargetId");
+    this.getDevices();
+    this.getDatatarget();
+    this.getPayloadDecoders();
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.forEach(s => s?.unsubscribe());
+  }
+
+  getDevices() {
+    const deviceSubscription = this.applicationService
+      .getApplication(this.applicationId)
+      .subscribe((application: Application) => {
+        this.iotDevices = application.iotDevices.sort((a, b) => a.name.localeCompare(b.name, "en", { numeric: true }));
+      });
+
+    this.subscriptions.push(deviceSubscription);
+  }
+
+  getLastPackage(event: MatSelectChange) {
+    this.iotDeviceId = event.value;
+    const deviceSub = this.deviceService.getIoTDevice(event.value).subscribe((device: IotDevice) => {
+      if (device.latestReceivedMessage) {
+        this.payloadData = JSON.stringify(device.latestReceivedMessage.rawData, null, 2);
+      }
+    });
+
+    const connectionSub = this.payloadDeviceDataTargetService.getByDataTarget(this.dataTargetId).subscribe(c => {
+      const connection = c.data.find(x => x.iotDevices.some(d => d.id === event.value));
+      this.payloadDecoderId = connection.payloadDecoder.id;
+    });
+
+    this.subscriptions.push(deviceSub, connectionSub);
+  }
+
+  pingDataTarget(dataPackage?: string) {
+    this.loading = true;
+    this.dataTargetService
+      .testDataTarget({
+        dataTargetId: this.dataTargetId,
+        iotDeviceId: this.iotDeviceId,
+        payloadDecoderId: this.payloadDecoderId,
+        dataPackage: dataPackage,
+      })
+      .subscribe(
+        response => {
+          this.testResponse = response?.result ? JSON.stringify(response.result, null, 2) : "";
+          this.decodedData = response?.decodedPayload ? JSON.stringify(response.decodedPayload, null, 2) : "";
+        },
+        error => {
+          console.log(error);
+        },
+        () => {
+          this.loading = false;
+        }
+      );
+  }
+
+  private getDatatarget() {
+    const dataTargetSubscription = this.dataTargetService.get(this.dataTargetId).subscribe((datatarget: Datatarget) => {
+      this.dataTarget = datatarget;
+    });
+
+    this.subscriptions.push(dataTargetSubscription);
+  }
+
+  private getPayloadDecoders() {
+    const sub = this.payloadDecoderService.getMultiple(100, 0, "asc", "name").subscribe(({ data }) => {
```

Comment:
Scenarie hvor man har over 100 payload decodes - er der så mulighed for at hente alle? Eller antages det, at der aldrig vil være over 100 decoders?

Reply:
Hæver limit til 1000, kan også godt lave en get minimal til backenden der kun henter navn og id, og alle i et kald